abstractions/typescript/src/dateOnly.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Represents a date only. ISO 8601.
+ */
+export class DateOnly implements DateOnlyInterface {
+    /**
+     * Creates a new DateOnly from the given string.
+     * @returns The new DateOnly
+     * @throws An error if the year is invalid
+     * @throws An error if the month is invalid
+     * @throws An error if the day is invalid
+     */
+    public constructor(
+        {year = 0,
+        month = 1,
+        day = 1} : Partial<DateOnlyInterface>
+    ) {
+        if (year < 0)
+            throw new Error("Year must be positive");
+        if (month < 1 || month > 12)
+            throw new Error("Month must be between 1 and 12");
+        if (day < 1 || day > 31)
+            throw new Error("Day must be between 1 and 31");
+        this.year = year;
+        this.month = month;
+        this.day = day;
+    }
```

Comment:
Would it be better to have a native `Date` instance to back this at initialization? In my head, it looks like it might be possible to pass in a date of 30 February 2020 which the `Date` instance would ideally flag.

Reply:
solved! thanks for catching this!